src/PropertyBase.cc

```diff
@@ -108,7 +108,8 @@ namespace ChimeraTK {
     if(handleLocking) {
       getEqFct()->unlock();
     }
-    for(const auto& prop : otherPropertiesToUpdate) {
+    for(const auto& weakProp : otherPropertiesToUpdate) {
+      auto prop = weakProp.lock();
```

Comment:
You are not checking that the locking worked. Is this always guaranteed?

Reply:
Here it's not guaranteed, I added a check.